libadikted/Makefile.win

```diff
@@ -1,106 +1,99 @@
-	$(CC) -c msg_log.c -o msg_log.o $(CFLAGS)
-
-arr_utils.o: arr_utils.c
-	$(CC) -c arr_utils.c -o arr_utils.o $(CFLAGS)
-
-lbfileio.o: lbfileio.c
-	$(CC) -c lbfileio.c -o lbfileio.o $(CFLAGS)
-
-adikted_private.res: adikted_private.rc 
-	$(WINDRES) -i adikted_private.rc --input-format=rc -o adikted_private.res -O coff 
+# Makefile created by KeeperFX/dkfans team
+
+# To create a debug executable, run with the command:
+# make DEBUG (THIS IS NOT IMPLEMENTED)
+
+# Instructions:
+# To create adikted.dll, run:
+# make all
+# To create adikited.so, run:
+# make all BUILD_FOR_LINUX=1
+# To clean up the built files (objects, library, and resource file) run:
+# make clean
+
+# Tested and working with:
+# - MSYS2 (MinGW 32-bit) - make
+# - MSYS2 (MinGW 32-bit) - mingw32-make
+# - MinGW (MSYS) - make
+
+# Executable files extension on host environment
+ifneq (,$(findstring Windows,$(OS)))
+  CROSS_EXEEXT = .exe
+else
+  CROSS_EXEEXT =
+  CROSS_COMPILE = i686-w64-mingw32-
+endif
+# Executable files extension on target environment (
+#BUILD_FOR_LINUX = 1
+ifneq (,$(BUILD_FOR_LINUX))
+  LIBEXT = .so
+  OTHERLIBEXT = .dll
+else
+  LIBEXT = .dll
+  OTHERLIBEXT = .so
+endif
+# Names of utility commands
+CPP                = $(CROSS_COMPILE)g++$(CROSS_EXEEXT) -std=c89
+CC                 = $(CROSS_COMPILE)gcc$(CROSS_EXEEXT) -std=c89 -Wall -Wextra -Wno-pedantic -Wno-conversion -Wno-traditional-conversion -Wno-sign-compare
+CFLAGS             = -c -O -I. -DBUILD_DLL -fexpensive-optimizations -O3 -march=pentium-mmx -mmmx
+CFLAGS-DEBUG       = -c -g -I. -DBUILD_DLL -fexpensive-optimizations -O3 -march=pentium-mmx -mmmx
+LINK               = $(CROSS_COMPILE)gcc$(CROSS_EXEEXT)
+WINDRES            = $(CROSS_COMPILE)windres$(CROSS_EXEEXT)
+RES                = obj/adikted_private.res
+RC                 = adikted_private.rc
+LFLAGS             = -shared
+LFLAGS-DEBUG       = -shared
+LIBS               = -L.
+SRC                = \
+arr_utils.c \
+bulcommn.c \
+dernc.c \
+draw_map.c \
+graffiti.c \
+graffiti_font.c \
+lbfileio.c \
+lev_column.c \
+lev_data.c \
+lev_files.c \
+lev_script.c \
+lev_things.c \
+libadi_main.c \
+memfile.c \
+msg_log.c \
+obj_actnpts.c \
+obj_column.c \
+obj_column_def.c \
+obj_column_per.c \
+obj_slabs.c \
+obj_things.c \
+xcubtxtr.c \
+xtabdat8.c \
+xtabjty.c
+OBJ                = $(addprefix obj/,$(SRC:.c=.o))
+BIN                = bin/adikted$(LIBEXT)
+DLLWRAP            = dllwrap$(CROSS_EXEEXT)
+DEF                = obj/libadikted.def
+STATICLIB          = obj/libadikted.a
+DLLLIBS            = --no-export-all-symbols --add-stdcall-alias  -march=pentium-mmx -mmmx
+RM                 = rm -f
+MKDIR              = mkdir -p
+
+.PHONY: pre-build post-build all clean
+
+all: pre-build $(SRC) $(BIN) post-build
+
+pre-build:
+	$(MKDIR) bin obj
+
+clean:
+	$(RM) $(OBJ) $(BIN) $(BIN:$(LIBEXT)=$(OTHERLIBEXT)) $(RES) $(DEF) $(STATICLIB)
+
+$(BIN): $(OBJ) $(RES)
+	$(DLLWRAP) --output-def $(DEF) --implib $(STATICLIB) $(OBJ) $(DLLLIBS) -o $(BIN) $(RES)
```

Comment:
This is how the old `Makefile.win` created the DLL

Reply:
dllwrap is deprecated and Windows specific, so I'd suggest going with how `Makefile` does it